src/main/scala/com/supersonic/datadog/TimeboardJSON.scala

```diff
@@ -42,16 +42,31 @@ object TimeboardJSON {
   }
 
   implicit val requestEncoder: Encoder[Request] = Encoder.instance { request =>
-    Json.obj(
-      "q" := request.series.map(_.render).mkString(", "),
-      "style" := request.style,
-      "type" := request.visualizationType)
+    val metadataList = request.series.flatMap { series =>
+      series.metadata.map(data => (series.series.render, data))
+    }
+
+    metadataList match {
+      case Nil => Json.obj(
```

Comment:
The code between the branches is almost duplicated just because of the single `metadata` field.
You can remove the duplication by doing something like the following:
```scala
val metadataList = request.series.flatMap { series =>
  series.metadata.map(data => (series.series.render, data))
}

val metadataField =
  if (metadataList.isEmpty) None
  else Some("metadata" := metadataList.toMap)

val fields = List(
  "q" := request.series.map(_.series.render).mkString(", "),
  "style" := request.style,
  "type" := request.visualizationType) ++ metadataField

Json.fromFields(fields)
```

Reply:
I like it, Done